### PR TITLE
Upgrade the packages, libraries to the latest.

### PR DESCRIPTION
**Pull request by Abhay5855**
Date: 13-06-2024

Developer Name: Abhay Patil

---

## Issue Ticket Number

https://github.com/Real-Dev-Squad/website-backend/issues/2048

## Description

Upgraded the packages and libraries to the latest and stable version including Node Js version to the latest.

### Documentation Updated?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [ ] Yes
- [x] No

## Test Coverage
<details>
<summary>Screenshot 1</summary>

image

</details>

## Additional Notes

- I have not upgraded the eslint and eslint related packages to the latest version v9 the reasons are the package  has been released few days back and is not a stable version yet.
Reference links  - https://www.npmjs.com/package/eslint
- I have removed the `eslint-plugin-standard` as it is now depricated and not used further.
link - https://www.npmjs.com/package/eslint-plugin-standard

### CODE REVIEW

**Review by Achintya-Chatterjee — CHANGES_REQUESTED** [review]
@Abhay5855 31 tests are failing. kindly look into that

**Comment by Abhay5855** [comment]
> @Abhay5855 31 tests are failing. kindly look into that

I looked into it they are also breaking on the develop branch, does it have any other setup that needs to be done, I checked running the tests on develop branch also

**Review by Achintya-Chatterjee — CHANGES_REQUESTED** [review]
@Abhay5855 one test is still failing.

**Comment by Abhay5855** [comment]
> @Abhay5855 one test is still failing.

Resolved this all tests are passing now

File: controllers/cloudflareCache.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ const purge = async (res, id, username, isSuperUser, userDetails) => {
         await logsQuery.addLog(
           logType.CLOUDFLARE_CACHE_PURGED,
           { userId: id, purgedFor: userDetails.user.id },
-          { message: "Cache Purged" }
+          { message: "Cache Purged" },
```
Comment (iamitprakash) [inline controllers/cloudflareCache.js:56]: do we really need this ","

File: controllers/discordactions.js [hunk 2]
```diff
@@ -282,7 +282,7 @@ const updateDiscordNicknames = async (req, res) => {
         } catch (error) {
           logger.error(`error getting user with matching discordId ${error.message}`);
         }
-      })
+      }),
```
Comment (iamitprakash) [inline controllers/discordactions.js:285]: same here

File: controllers/discordactions.js [hunk 3]
```diff
@@ -366,7 +366,7 @@ const syncDiscordGroupRolesInFirestore = async (req, res) => {
             {
               roleid: role.id,
             },
-            roleInFirestore.id
+            roleInFirestore.id,
```
Comment (iamitprakash) [inline controllers/discordactions.js:369]: ,

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: controllers/cloudflareCache.js (hunk 1 above)
Comment (Abhay5855) [reply controllers/cloudflareCache.js:56]: It's Due to new prettier update, prettier is adding that , while fixing the linting error, else it's throwing linting error @iamitprakash

File: controllers/discordactions.js (hunk 2 above)
Comment (Abhay5855) [reply controllers/discordactions.js:285]: These file changes are due to the linting fixes of new prettier package upgrade.

Comment (Abhay5855) [reply controllers/discordactions.js:285]: image

**Comment by prakashchoudhary07** [comment]
66 files changes to update packages?

**Comment by prakashchoudhary07** [comment]
Please keep the formatting separate from he update changes

**Comment by Abhay5855** [comment]
> Please keep the formatting separate from he update changes

@prakashchoudhary07 Made the changes, Not including the formatting changes in this, can you review again.

File: controllers/cloudflareCache.js (hunk 1 above)
Comment (Abhay5855) [reply controllers/cloudflareCache.js]: No, it is not needed as such as prettier is already covering that rule, Made the changes can you review this again. @iamitprakash.

**Comment by Abhay5855** [comment]
> Please keep the formatting separate from he update changes

Done kept it seperate

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**